### ISSUE TITLE
source.remaining returns (undef)

### ISSUE THREAD

**Issue by ScuttleSE** [savonet/liquidsoap#2255]
Running 2.0.3, and when I telnet into the server and run f.ex 96.remaining, I get (undef) as the result

Everything else works just fine, I just can't seem to get the remaining time on the currently playing song...

```
set("harbor.bind_addr","0.0.0.0")
set("server.telnet.bind_addr", "0.0.0.0")
set("server.telnet",true)
set("log.file",false)
set("log.stdout",true)
set("log.level",3)

def get_request() =
  result = process.read.lines("python3 /home/liq/getsong.py")
  log("next song: #{result}")
  request.create(list.hd(default="", result))
end

def auth_gbsfm(params)
  user = params.user
  password = params.password
  ret = process.read.lines("python3 /home/liq/streampw.py #{password}")
  ret = list.hd(default="",ret)
  if ret == "true" then
    true
  else
    false
  end
end

#Jingles, an internal playlist we interact with via telnet
jingles = request.queue(id="jingles")

#Live source
live = input.harbor("gbsfm_live",port=4457,auth=auth_gbsfm)

#Website playlist source
gbsfm = request.dynamic(prefetch=0,  id="gbsfm", get_request)

if not gbsfm.add(get_request()) then
  log("Error while queueing initial request!")
end

def on_end(_, _) =
  new_request = get_request()
  if not gbsfm.add(new_request) then
    log("Error while queueing next request!")
  end
end

gbsfm = source.on_end(gbsfm, on_end)

#Fallback
beees = single("/home/liq/bees.mp3")

#Make sure mono files are being played
gbsfm = audio_to_stereo(gbsfm)

#Put them all together
gbsfm = fallback(track_sensitive=false, [live, gbsfm, beees])

#And now we add the jingles
gbsfm = smooth_add(normal=gbsfm,special=jingles)

output.icecast(
    %mp3(bitrate=96),
    fallible = true,
    host = "localhost",
    port = 8000,
    password = "xxx",
    mount = "96", gbsfm)

output.icecast(
    %mp3(bitrate=128),
    fallible = true,
    host = "localhost",
    port = 8000,
    password = "xxx",
    mount = "128", gbsfm)

output.icecast(
    %mp3(bitrate=192),
    fallible = true,
    host = "localhost",
    port = 8000,
    password = "xxx",
    mount = "192", gbsfm)

output.icecast(
    %mp3(bitrate=256),
    fallible = true,
    host = "localhost",
    port = 8000,
    password = "xxx",
    mount = "256", gbsfm)

output.icecast(
    %mp3(bitrate=320),
    fallible = true,
    host = "localhost",
    port = 8000,
    password = "xxx",
    mount = "320", gbsfm)

output.icecast(
    %ogg(%flac),
    fallible = true,
    host = "localhost",
    port = 8000,
    password = "xxx",
    mount = "flac", gbsfm)

# dump live dj recordings to a file
timestamp = '%Y%m%d%H%M%S'
show_title = 'GBSFM_live_show'
output.file(
    %mp3(bitrate=320, id3v2=true),
    fallible=true,
    reopen_on_metadata=false,
    "/srv/music/liveshows/#{show_title} Recorded On #{timestamp}.mp3",
    live)

```

**Comment by ScuttleSE** [comment]
It seems to have something to do with smooth_add

Once I removed that and added the jingle source like this instead, the remaining time seems to work fine

`gbsfm = fallback(track_sensitive=false, [live, jingles, gbsfm, beees])`

**Comment by toots** [comment]
Thanks for reporting. Live source will always have undefined remaining so you should expect it when it's being played. However, the `add` operator being pretty conservative situations with undefined remaining and remaining in other sources, assuming that the overall remaining was undefined in these cases. I have changed it to match the min of the remaining time over all sources that have non undefined remaining time. This is a little optimistic but I think this corresponds to most people's expectation.

Fix has been pushed to `v2.0.4-preview` and `main`.
